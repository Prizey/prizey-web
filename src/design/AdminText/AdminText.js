import React from 'react'
import { Info } from 'croods'

const AdminText = ({ render, tags = ''}) => {
  console.log('Tags ->', tags)
  return (
  <Info
    id={tags}
    name="adminText"
    path={`/admin_texts?tags[]=${tags}`}
    render={render}
    disableCache
  />
)}

export default AdminText
