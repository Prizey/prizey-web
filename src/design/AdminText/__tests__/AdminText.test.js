import React from 'react'
import renderer from 'react-test-renderer'
import AdminText from '../AdminText'

jest.mock('croods', () => ({
  Info: ({ render, parseResponse, ...props }) => (
    <div {...props}>Info - {render()}</div>
  ),
}))

describe('when the component is mounted', () => {
  it('renders correctly', () => {
    const params = {
      render: () => <div>the result</div>,
      tags: 'homepage_cta',
    }

    const tree = renderer.create(<AdminText {...params} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('test the parse response', () => {
    const params = {
      render: () => <div>the result</div>,
      tags: 'homepage_cta',
    }
    const tree = renderer.create(<AdminText {...params} />).root
    const component = tree.findByProps({ name: 'adminText' })

    const data = [{ homepageCta: 'content' }]
    const response = component.props.parseResponse(data)

    expect(response).toEqual({
      info: {
        homepageCta: 'content',
        id: 'homepage_cta',
      },
    })
  })
})

describe('with more than one tag', () => {
  it('renders the path correctly', () => {
    const params = {
      render: () => <div>the result</div>,
      tags: ['homepage_cta', 'foobar_text'],
    }

    const tree = renderer.create(<AdminText {...params} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
