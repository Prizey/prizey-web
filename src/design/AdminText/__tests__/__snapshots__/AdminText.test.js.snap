// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when the component is mounted renders correctly 1`] = `
<div
  disableCache={true}
  id="homepage_cta"
  name="adminText"
  path="/admin_texts?tags[]=homepage_cta"
>
  Info - 
  <div>
    the result
  </div>
</div>
`;

exports[`with more than one tag renders the path correctly 1`] = `
<div
  disableCache={true}
  id="homepage_cta,foobar_text"
  name="adminText"
  path="/admin_texts?tags[]=homepage_cta&tags[]=foobar_text"
>
  Info - 
  <div>
    the result
  </div>
</div>
`;
